client/wfnews-war/src/main/angular/src/app/components/report-of-fire/compass-page/rof-compass-page.component.ts

```diff
@@ -0,0 +1,94 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { RoFPage } from "../rofPage";
+import { ReportOfFire } from '../reportOfFireModel';
+import { CommonUtilityService } from "../../../services/common-utility.service";
+import { CapacitorService } from "../../../services/capacitor-service";
+
+@Component({
+  selector: 'rof-compass-page',
+  templateUrl: './rof-compass-page.component.html',
+  styleUrls: ['./rof-compass-page.component.scss'],
+  changeDetection: ChangeDetectionStrategy.Default
+})
+export class RoFCompassPage extends RoFPage implements OnInit {
+  public compassFaceUrl: string
+  public compassHandUrl: string
+  public compassHeading: number;
+  public currentLat: number
+  public currentLong: number
+  public heading: string = "0° N";
+
+  constructor(private commonUtilityService: CommonUtilityService,
+    private capacitorService: CapacitorService) {
+    super();
+  }
+  
+initialize (data: any, index: number, reportOfFire: ReportOfFire) {
+    super.initialize(data, index, reportOfFire);
+    this.compassFaceUrl = data.compassFaceUrl;
+    this.compassHandUrl = data.compassHandUrl;   
+  }
+
+ngOnInit(): void {
+  this.getOrientation(); 
+  this.useMyCurrentLocation();
+}
+
+getOrientation() {
+  if (this.capacitorService.isIOSPlatform) {
+    (DeviceOrientationEvent as any).requestPermission()
+      .then((response) => {
+        if (response === "granted") {
+          window.addEventListener("deviceorientation", this.handler, true);
+        } else {
+          alert("has to be allowed!");
```

Comment:
Should we use a popup instead of an alert?
The message is probably not appropriate. We should update to say something a little more user friendly "Location services are required" or something